src/arrayMethodSort.js

```diff
@@ -4,8 +4,36 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compareFn =
+      compareFunction ||
+      function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+
+        if (strA < strB) {
+          return -1;
+        }
+
+        if (strA > strB) {
+          return 1;
+        }
+
+        return 0;
+      };
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - i - 1; j++) {
+        if (compareFn(this[j], this[j + 1]) > 0) {
+          const temp = this[j];
+
+          this[j] = this[j + 1];
+          this[j + 1] = temp;
+        }
+      }
+    }
+
+    return this;
   };
```

Comment:
Modifying the `Array.prototype` can lead to potential issues, especially if other libraries or code rely on the standard behavior of arrays. Consider using a different approach, such as creating a utility function that accepts an array as an argument and returns a sorted array.